jax/_src/numpy/util.py

```diff
@@ -117,6 +119,22 @@ def _parse_extra_params(extra_params: str) -> dict[str, str]:
   return {p.partition(' : ')[0].partition(', ')[0]: p for p in parameters}
 
 
+def _place_array(x: Array, device: xc.Device | Sharding | None = None, copy=None) -> Array:
+  """Helper utility for copying an array, or placing it on a device or sharding.
+
+  This utility uses `jax.device_put` for device placement.
+  """
+
+  if device is not None:
```

Comment:
You don't need `device is not None` here? device_put accepts None has the device.

Reply:
`_place_array(x, copy=False)` should avoid the error, and `_place_array(x, copy=True)` should force a copy -- which afaik `device_put(x, None)` does not do, hence guarding for `device is not None`

Reply:
Using `device_put` regardless of `device` results in get `NotImplementedError: sparse rule for device_put is not implemented` errors for sparse matrices using `.astype`, so I'm undoing this change for now so we can avoid using `device_put` unless `device` is specified.